staging/src/k8s.io/client-go/tools/cache/listers.go

```diff
@@ -44,8 +51,15 @@ func ListAll(store Store, selector labels.Selector, appendFn AppendFunc) error {
 }
 
 func ListAllByNamespace(indexer Indexer, namespace string, selector labels.Selector, appendFn AppendFunc) error {
+	selectAll := selector.Empty()
 	if namespace == metav1.NamespaceAll {
 		for _, m := range indexer.List() {
+			if selectAll {
```

Comment:
Shouldn't the slow search without index block also have this optimization ?

Reply:
That should almost never happen (it's only error handling). But feel free to add that.